### PR TITLE
Display Partner Balance on homepage

### PR DESCRIPTION
**Pull request by corymcdonald**
Closes issue #1602

Things left to do

- [x] Determine if prettier should have run on the README 🤔 
- [x] Write some tests

### CODE REVIEW

File: app/models/partner.rb [hunk 1]
```diff
@@ -13,7 +13,26 @@ def ensure_role
...
     self[:name] || self[:email]
   end
+
+  private
+
+  def invoice_amount
+    invoices = Invoice.where(partner_id: id, paid: false)
+    amounts = invoices.map { |i| i.finalized_amount || i.amount }
+
+    amounts.map { |x| x.gsub(/,/, '').to_i }.reduce(:+)
```
Comment (dlipeles) [inline app/models/partner.rb]: Quick question, do we prefer .gsub or trim? I've been having good results with 
```ruby
str.tr(",", "")
```

But I'm not sure which is better

File: app/models/partner.rb [hunk 2]
```diff
@@ -13,7 +13,26 @@ def ensure_role
     self.role = 'partner'
   end
 
+  def balance
+    wallet.contribution_balance.amount_bat + invoice_amount
+  end
+
+  def balance_in_currency
+    wallet.contribution_balance.add_bat(invoice_amount)
+
```
Comment (dlipeles) [inline app/models/partner.rb]: nit: maybe we can rm the whitespace on `line 22`

File: app/models/invoice.rb [hunk 3]
```diff
@@ -24,6 +24,10 @@ class ReadOnlyError < StandardError; end
 
   validates :date, uniqueness: { scope: :partner_id }
 
+  # Ensure these two values are numbers even though field is a string
+  validates :amount,  numericality: true, allow_nil: true
+  validates :finalized_amount,  numericality: true, allow_nil: true
```
Comment (dlipeles) [inline app/models/invoice.rb]: Maybe we can even validate further with one of these

 https://guides.rubyonrails.org/active_record_validations.html#numericality

Perhaps ```:greater_than_or_equal_to: 0```

Not sure if that would clash with nil though, totally your call, just food for thought 🍕

File: app/models/partner.rb (hunk 1 above)
Comment (corymcdonald) [reply app/models/partner.rb]: https://github.com/brave-intl/publishers/pull/1667/commits/092d85279ad37ad6a8ce82db79c7972594e84ead

File: app/models/partner.rb (hunk 2 above)
Comment (corymcdonald) [reply app/models/partner.rb]: https://github.com/brave-intl/publishers/pull/1667/commits/092d85279ad37ad6a8ce82db79c7972594e84ead

File: app/models/invoice.rb (hunk 3 above)
Comment (corymcdonald) [reply app/models/invoice.rb]: Unfortunately we can't allow_nil if we do this 😢

File: app/models/partner.rb [hunk 4]
```diff
@@ -13,7 +13,25 @@ def ensure_role
...
+    wallet.contribution_balance.amount_default_currency
+  end
+
   def name
     self[:name] || self[:email]
   end
+
+  private
+
+  def invoice_amount
```
Comment (dlipeles) [inline app/models/partner.rb]: @yachtcaptain23 passed me this feedback the other day, I think it applies to this section 

consider 

```ruby 
def invoice_amount
    invoices = Invoice.where(partner_id: id, paid: false)
    amounts = invoices.map { |i| i.finalized_amount || i.amount }

    amounts.map { |x| x.tr(",", "").to_i }.reduce(:+) || 0
end
```

His rationale is prevent `add_bat` from throwing an error if the amount is `nil`

Comment (corymcdonald) [reply app/models/partner.rb]: Done! 
https://github.com/brave-intl/publishers/pull/1667/commits/91b10f64f07745e510dcba4d2625bb439f90a752

**Pull request merged**